core/Updater.php

```diff
@@ -233,37 +233,39 @@ public function hasMajorDbUpdate()
      */
     public function getSqlQueriesToExecute()
     {
-        $queries = array();
-        $classNames = array();
-
-        foreach ($this->componentsWithUpdateFile as $componentName => $componentUpdateInfo) {
-            foreach ($componentUpdateInfo as $file => $fileVersion) {
-                require_once $file; // prefixed by PIWIK_INCLUDE_PATH
-
-                $className = $this->getUpdateClassName($componentName, $fileVersion);
-                if (!class_exists($className, false)) {
-                    // throwing an error here causes Matomo to show the safe mode instead of showing an exception fatal only
-                    // that makes it possible to deactivate / uninstall a broken plugin to recover Matomo directly
-                    throw new \Error("The class $className was not found in $file");
-                }
+        return Access::doAsSuperUser(function() {
+            $queries    = [];
+            $classNames = [];
+
+            foreach ($this->componentsWithUpdateFile as $componentName => $componentUpdateInfo) {
+                foreach ($componentUpdateInfo as $file => $fileVersion) {
+                    require_once $file; // prefixed by PIWIK_INCLUDE_PATH
+
+                    $className = $this->getUpdateClassName($componentName, $fileVersion);
+                    if (!class_exists($className, false)) {
+                        // throwing an error here causes Matomo to show the safe mode instead of showing an exception fatal only
+                        // that makes it possible to deactivate / uninstall a broken plugin to recover Matomo directly
+                        throw new \Error("The class $className was not found in $file");
+                    }
 
-                if (in_array($className, $classNames)) {
-                    continue; // prevent from getting updates from Piwik\Columns\Updater multiple times
-                }
+                    if (in_array($className, $classNames)) {
+                        continue; // prevent from getting updates from Piwik\Columns\Updater multiple times
+                    }
 
-                $classNames[] = $className;
+                    $classNames[] = $className;
 
-                /** @var Updates $update */
-                $update = StaticContainer::getContainer()->make($className);
-                $migrationsForComponent = $update->getMigrations($this);
-                foreach ($migrationsForComponent as $index => $migration) {
-                    $migration = $this->keepBcForOldMigrationQueryFormat($index, $migration);
-                    $queries[] = $migration;
+                    /** @var Updates $update */
+                    $update                 = StaticContainer::getContainer()->make($className);
+                    $migrationsForComponent = $update->getMigrations($this);
```

Comment:
@sgiehl here we could still maybe narrow it down to only the `$update->getMigrations($this);` call maybe?

Reply:
done.